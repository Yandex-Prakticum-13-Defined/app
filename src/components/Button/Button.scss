.button {
  padding: 10px;
  box-sizing: border-box;
  border-radius: 15px;
  font-size: 30px;
  border: none;
  border-bottom: 3px solid var(--color-main);
  background: none;

  &:disabled {
    color: var(--button-disabled);
  }

  &:enabled {
    color: var(--light-blue);
  }

  &:enabled:hover {
    cursor: pointer;
    opacity: 0.5;
  }
}
