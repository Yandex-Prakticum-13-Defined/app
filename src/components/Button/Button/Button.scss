.button {
  width: 70%;
  padding: 15px;
  text-align: center;
  cursor: pointer;
  border-radius: 15px;
  font-weight: 500;
  font-size: 30px;
  line-height: 11px;
  text-decoration: none;
  border: none;
  outline: none;
  border-bottom: 3px solid black;
  background: none;
}
