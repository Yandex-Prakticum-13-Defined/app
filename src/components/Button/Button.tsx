import React from 'react';
import './Button.scss';

interface IButton {
  className?: string;
  onClick?: any;
  title?: string;
  type?: 'button' | 'submit' | 'reset' | undefined;
  disabled?: boolean
}

const Button = ({
  className,
  onClick,
  title,
  type,
  disabled
}: IButton) => (
  <button className={className || 'button'} onClick={onClick} type={type} disabled={disabled}>
    {title}
  </button>
);

export default Button;
