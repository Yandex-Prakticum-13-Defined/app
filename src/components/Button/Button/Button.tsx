import React from 'react';
import './Button.scss'

interface IButton {
  className?: string;
  onClick?: () => void;
  title?: string;
  type?: "button" | "submit" | "reset" | undefined;
}

const Button = ({
                  className,
                  onClick,
                  title,
                  type,
                }: IButton) => {
  return (
    <button className={className ? className : 'button'} onClick={onClick} type={type}>
      {title}
    </button>
  )
}

export default Button;
