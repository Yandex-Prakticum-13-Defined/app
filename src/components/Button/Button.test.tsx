import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from './Button';

describe('Тестируем Button.tsx', () => {
  it('проверяем снимок', () => {
    const buttonComponent = renderer.create(<Button type='button'/>).toJSON();
    expect(buttonComponent).toMatchSnapshot();
  });

  it('содержит текст', () => {
    const { getByText } = render(<Button type='button' title='Тестовая кнопка'/>);
    expect(getByText('Тестовая кнопка')).toBeInTheDocument();
  });

  it('активна по умолчанию', () => {
    const { getByText } = render(<Button type='button' title='Тестовая кнопка'/>);
    expect(getByText('Тестовая кнопка')).toBeEnabled();
  });

  it('заблокирована при передаче disabled', () => {
    const { getByText } = render(<Button type='button' title='Тестовая кнопка' disabled/>);
    expect(getByText('Тестовая кнопка')).toBeDisabled();
  });

  it('вызывается событие click', async () => {
    const handleClick = jest.fn();
    const { getByText } = render(<Button type='button' title='Тестовая кнопка' onClick={handleClick}/>);
    const user = userEvent.setup();
    await user.click(getByText('Тестовая кнопка'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
