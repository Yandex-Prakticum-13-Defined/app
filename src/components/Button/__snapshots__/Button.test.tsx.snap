// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем Button.tsx проверяем снимок 1`] = `
<button
  className="button"
  type="button"
/>
`;
